conf/nomad/presto.hcl

```
@@ -119,47 +166,214 @@ job "${nomad_job_name}" {
+          # Custom certificate authenticator configuration
+          # TODO: add variable and make it optional
+          "local/presto/plugin/presto-consul-connect.jar:/usr/lib/presto/plugin/consulconnect/presto-consul-connect.jar",
+          "local/presto/certificate-authenticator.properties:/lib/presto/default/etc/certificate-authenticator.properties",
+          # Mount for debug purposes
+          %{ if debug }"local/presto/log.properties:/lib/presto/default/etc/log.properties",%{ endif }
+          # Hive connector configuration
+          "local/presto/hive.properties:/lib/presto/default/etc/catalog/hive.properties",
+          # Mounting /local/hosts to /etc/hosts overrides default docker mount.
+          "local/hosts:/etc/hosts",
+          # Mounting modified entrypoint.
+          # The will start a background process to update /etc/hosts from /local/hosts on a 2 second interval
+          # This is needed in order for the workers and coordinators to communicate by name
+          "local/scripts/run-presto:/usr/lib/presto/bin/run-presto",
         ]
       }
+      artifact {
+        # Download custom certificate authenticator plugin
+        # TODO: add variable for this and make it optional
+        source = "https://oss.sonatype.org/service/local/repositories/releases/content/io/github/gugalnikov/presto-consul-connect/2.1.0/presto-consul-connect-2.1.0-jar-with-dependencies.jar"
+        mode = "file"
+        destination = "local/presto/plugin/presto-consul-connect.jar"
+      }
       template {
         data = <<EOH
-          MINIO_ACCESS_KEY = "${minio_access_key}"
-          MINIO_SECRET_KEY = "${minio_secret_key}"
+          connector.name=hive-hadoop2
+          hive.s3.aws-access-key=${minio_access_key}
+          hive.s3.aws-secret-key=${minio_secret_key}
+          hive.s3.endpoint=http://{{ env "NOMAD_UPSTREAM_ADDR_${minio_service_name}" }}
+          hive.metastore.uri=thrift://{{ env "NOMAD_UPSTREAM_ADDR_${hivemetastore_service_name}" }}
+          hive.s3select-pushdown.enabled=true
+          hive.non-managed-table-writes-enabled=true
+          hive.s3.max-connections=5000
+          hive.s3.max-error-retries=100
+          hive.s3.socket-timeout=31m
+          hive.s3.ssl.enabled=false
+          hive.metastore-timeout=1m
+          hive.s3.path-style-access=true
           EOH
+        destination = "/local/presto/hive.properties"
+      }
+      template {
+        # TODO: Create issue with hashicorp. Is there a way to mount directly to /etc/hosts ( for continual updates )
+        data = <<EOF
+127.0.0.1 %{ if node_type == "coordinator" }${service_name}%{ else }{{env "NOMAD_PORT_connect"}}.${service_name}-worker%{ endif } localhost
+{{ range services }}
+{{- if .Name | regexMatch "${service_name}"}}{{ range connect .Name }}{{.Address}} {{.Name}}
+{{end}}{{end}}
+{{- if .Name | regexMatch "${service_name}-worker"}}{{ range connect .Name }}{{.Address}} {{.Port}}.{{.Name}}
+{{end}}{{end}}{{end}}
+EOF
```

Could you add a quick description here? 
```suggestion
        # This will add all hosts with service names presto-worker and presto to the hosts file.
        data = <<EOF
127.0.0.1 %{ if node_type == "coordinator" }${service_name}%{ else }{{env "NOMAD_PORT_connect"}}.${service_name}-worker%{ endif } localhost
{{ range services }}
{{- if .Name | regexMatch "${service_name}"}}{{ range connect .Name }}{{.Address}} {{.Name}}
{{end}}{{end}}
{{- if .Name | regexMatch "${service_name}-worker"}}{{ range connect .Name }}{{.Address}} {{.Port}}.{{.Name}}
{{end}}{{end}}{{end}}
EOF
```

---

Yes.